File: java/socketd/src/main/java/org/noear/socketd/client/ClientConnector.java
package org.noear.socketd.client;

import org.noear.socketd.core.Channel;
import org.noear.socketd.core.HeartbeatHandler;

import java.io.Closeable;

/**
 * 客户端连接器
 *
 * @author noear
 * @since 2.0
 */
public interface ClientConnector extends Closeable {
    /**
     * 心跳处理
     */
    HeartbeatHandler heartbeatHandler();

    /**
     * 心跳频率（单位：毫秒）
     */
    long heartbeatInterval();

    /**
     * 是否自动重连
     */
    boolean autoReconnect();

    /**
     * 最大允许请求数（用于背压控制）
     */
    int maxRequests();

    /**
     * 连接
     *
     * @return 通道
     */
    Channel connect() throws Exception;
}

File: javascript/socketd-examples/demo01-js-ws-esm/js/socketd/transport/client/ClientConnector.js
/**
 * 客户端连接器基类
 *
 * @author noear
 * @since 2.0
 */
export class ClientConnectorBase {
    constructor(client) {
        this._client = client;
    }
    heartbeatHandler() {
        return this._client.getHeartbeatHandler();
    }
    heartbeatInterval() {
        return this._client.getHeartbeatInterval();
    }
    autoReconnect() {
        return this._client.getConfig().isAutoReconnect();
    }
}
